rskj-core/src/main/java/co/rsk/peg/Bridge.java

```diff
@@ -1094,20 +1095,20 @@ public boolean hasBtcBlockCoinbaseTransactionInformation(Object[] args) {
         return bridgeSupport.hasBtcBlockCoinbaseTransactionInformation(blockHash);
     }
 
-    public int registerBtcTransfer(Object[] args) {
-        logger.trace("registerBtcTransfer");
-
-        byte[] btcTxSerialized = (byte[]) args[0];
-        int height = ((BigInteger) args[1]).intValue();
-        byte[] pmtSerialized = (byte[]) args[2];
-        Sha256Hash derivationArgumentsHash = Sha256Hash.wrap((byte[]) args[3]);
-        Address userRefundAddress = new Address(bridgeConstants.getBtcParams(), (byte[]) args[4]);
-        RskAddress lbcAddress = new RskAddress((String) args[5]);
-        Address lpBtcAddress = new Address(bridgeConstants.getBtcParams(), (byte[]) args[6]);
-        boolean executionStatus = (Boolean) args[7];
+    public int registerFastBridgeBtcTransaction(Object[] args) {
+        logger.trace("registerFastBridgeBtcTransaction");
 
         try {
-            return bridgeSupport.registerBtcTransfer(
+            byte[] btcTxSerialized = (byte[]) args[0];
+            int height = ((BigInteger) args[1]).intValue();
+            byte[] pmtSerialized = (byte[]) args[2];
+            Sha256Hash derivationArgumentsHash = Sha256Hash.wrap((byte[]) args[3]);
+            Address userRefundAddress = new Address(bridgeConstants.getBtcParams(), (byte[]) args[4]);
+            RskAddress lbcAddress = new RskAddress(((DataWord)args[5]).getLast20Bytes());
```

Comment:
Coukld we check if the argument is really an address? Now, the code discard the "left-most" bytes, that could contain anything.

Reply:
I don't expect more than 20 bytes, since the "address" type of solidity is internally a 20-byte array. When DataWord is cast for handling, it is converted to a 32-byte array by setting 0s to it. It may be that I am not seeing something that you are.

Reply:
I will modify it! thank you!